js/packages/openinference-core/src/trace/trace_config/maskingRules.ts

```diff
@@ -0,0 +1,199 @@
+import { AttributeValue } from "@opentelemetry/api";
+import { REDACTED_VALUE } from "./constants";
+import { SemanticConventions } from "@arizeai/openinference-semantic-conventions";
+import { MaskingRule, MaskingRuleArgs } from "./types";
+
+/**
+ * Masks (redacts) input text in LLM input messages.
+ * Will mask information stored under the key `llm.input_messages.[i].message.content`.
+ * * @example
+ * ```typescript
+ *  maskOutputTextRule.condition({
+ *      config: {hideInputText: true},
+ *      key: "llm.input_messages.[i].message.content"
+ *  }) // returns true so the rule applies and the value will be redacted
+ */
+const maskInputTextRule: MaskingRule = {
```

Comment:
pulled more complicated masking rules out  to write doc comments and wrote more simple ones inline (in the array below)

Reply:
this rule is based on the same config value (hideINputText) as the rule on line 55, they can be combined into one rule if we think that it's better to have one rule per field, this matches the python implementation, but i think it might be better to merge them